src/python/pants/rules/core/lint.py

```diff
@@ -25,16 +25,32 @@ class Lint(Goal):
   # Blocked on https://github.com/pantsbuild/pants/issues/8351
   name = 'lint-v2'
 
+  @classmethod
+  def register_options(cls, register):
+    super().register_options(register)
+    register('--transitive', type=bool, default=True,
```

Comment:
Rather than adding this option in multiple places, is it worth extracting a mix-in?